2/Dockerfile

```diff
@@ -1,21 +1,41 @@
 FROM buildpack-deps:jessie
 
-# remove several traces of debian python
-RUN apt-get purge -y python.*
+# ensure local pypy is preferred over distribution python
+ENV PATH /usr/local/bin:$PATH
 
 # http://bugs.python.org/issue19846
 # > At the moment, setting "LANG=C" on a Linux system *fundamentally breaks Python 3*, and that's not OK.
 ENV LANG C.UTF-8
 
-ENV PYPY_VERSION 5.4.0
+# runtime dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+		tcl \
+		tk \
```

Comment:
Note that I'm not installing these in the slim images, as the same is not done in the Python slim images.

You can check for the necessary libraries doing something like:
```shell
find /usr/local -type f -iname '*.so' | xargs ldd | grep 'not found'
```

On the non-slim Python/PyPy images this returns nothing. On the slim PyPy images this returns:
```
        libtcl8.6.so => not found
       	libtk8.6.so => not found
       	libgdbm.so.3 => not found
```

On the slim Python images it returns:
```
       	libtk8.6.so => not found
       	libtcl8.6.so => not found
       	libX11.so.6 => not found
```

Not sure if missing gdbm is an issue...